presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergUtil.java

```diff
@@ -887,12 +888,17 @@ public DeleteFile next()
             return result;
         }
 
-        private boolean shouldIncludeFile(DeleteFile file)
+        private boolean shouldIncludeFile(DeleteFile file, PartitionSpec partitionSpec)
         {
             boolean matchesPartition = !requestedPartitionSpec.isPresent() ||
                     requestedPartitionSpec.get().equals(partitionSpecsById.get(file.specId()).fields().stream().map(PartitionField::fieldId).collect(Collectors.toSet()));
```

Comment:
nit: no need for another lookup in `partitionSpecsById` since it is now passed in as an arg

Reply:
Nice point, fixed!